src/TurbulenceClosures/turbulence_closure_implementations/convective_adjustment_vertical_diffusivity.jl

```diff
@@ -0,0 +1,220 @@
+    background_κz = tracer_diffusivities(tracers, closure.background_κz)
+    convective_κz = tracer_diffusivities(tracers, closure.convective_κz)
+    return ConvectiveAdjustmentVerticalDiffusivity{TD}(convective_κz,
+                                                       closure.convective_νz,
+                                                       background_κz,
+                                                       closure.background_νz)
+end
+
+
+function DiffusivityFields(arch, grid, tracer_names, bcs, closure::CAVD)
+    data = new_data(Bool, arch, grid, (Center, Center, Face))
+    stable_buoyancy_gradient = Field(Center, Center, Face, arch, grid, nothing, data)
+    return (; stable_buoyancy_gradient)
+end       
+
+function calculate_diffusivities!(diffusivities, arch, grid, closure::CAVD, buoyancy, velocities, tracers)
+
+    event = launch!(arch, grid, :xyz,
+                    compute_stability!, diffusivities, grid, tracers, buoyancy,
+                    dependencies = device_event(arch))
+
+    wait(device(arch), event)
+
+    return nothing
+end
+
+@inline is_stableᶜᶜᶠ(i, j, k, grid, tracers, buoyancy) = ∂z_b(i, j, k, grid, buoyancy, tracers) > 0
+
+@kernel function compute_stability!(diffusivities, grid, tracers, buoyancy)
+    i, j, k, = @index(Global, NTuple)
+    @inbounds  diffusivities.stable_buoyancy_gradient[i, j, k] = is_stableᶜᶜᶠ(i, j, k, grid, tracers, buoyancy)
+end
+
+#####
+##### Fluxes
+#####
+
+# u
+@inline function νᶠᶜᶠ(i, j, k, grid, clock, closure::CAVD, stable_buoyancy_gradient)
+    @inbounds stableᶠᶜᶠ = stable_buoyancy_gradient[i, j, k] || stable_buoyancy_gradient[i+1, j, k]
+
+    ν = ifelse(stableᶠᶜᶠ,
+               νᶠᶜᶠ(i, j, k, grid, clock, closure.background_νz),
+               νᶠᶜᶠ(i, j, k, grid, clock, closure.convective_νz))
+
+    return ν
+end
+
+@inline function viscous_flux_uz(i, j, k, grid, closure::CAVD, clock, velocities, diffusivities, tracers, buoyancy)
+    ν = νᶠᶜᶠ(i, j, k, grid, clock, closure, diffusivities.stable_buoyancy_gradient)
+    return - ν * ∂zᵃᵃᶠ(i, j, k, grid, velocities.u)
+end
+
+# v
+@inline function νᶜᶠᶠ(i, j, k, grid, clock, closure::CAVD, stable_buoyancy_gradient)
+    @inbounds stableᶜᶠᶠ = stable_buoyancy_gradient[i, j, k] || stable_buoyancy_gradient[i, j+1, k]
+
+    ν = ifelse(stableᶜᶠᶠ,
+               νᶜᶠᶠ(i, j, k, grid, clock, closure.background_νz),
+               νᶜᶠᶠ(i, j, k, grid, clock, closure.convective_νz))
+
+    return ν
+end
+
+@inline function viscous_flux_vz(i, j, k, grid, closure::CAVD, clock, velocities, diffusivities, tracers, buoyancy)
+    ν = νᶜᶠᶠ(i, j, k, grid, clock, closure, diffusivities.stable_buoyancy_gradient)
+    return - ν * ∂zᵃᵃᶠ(i, j, k, grid, velocities.v)
+end
+
+# w
+@inline function νᶜᶜᶜ(i, j, k, grid, clock, closure::CAVD, stable_buoyancy_gradient)
+    @inbounds stableᶜᶜᶜ = stable_buoyancy_gradient[i, j, k] || stable_buoyancy_gradient[i, j, k+1]
```

Comment:
and here?

Reply:
Now this one I think is right. But on the others, I agree with you!

Reply:
to explain: for the others we evaluate an object at centers to faces in i or j respectively. here, we evaluate an object located at faces to cell centers --- the reverse logic